pulsar-storm/src/main/java/org/apache/pulsar/storm/PulsarBolt.java

```diff
@@ -162,11 +156,6 @@ public void declareOutputFields(OutputFieldsDeclarer declarer) {
         pulsarBoltConf.getTupleToMessageMapper().declareOutputFields(declarer);
     }
 
-    protected static boolean isTickTuple(Tuple tuple) {
-        return tuple != null && Constants.SYSTEM_COMPONENT_ID.equals(tuple.getSourceComponent())
```

Comment:
is there any behavior change for `TickTuple` in newer version of storm?

Reply:
There is not

Reply:
Since the Tick tuple frequency is not set anywhere, the tick tuples are not enabled, thus I don't think this is neccessary

Reply:
don't we set `Config.TOPOLOGY_TICK_TUPLE_FREQ_SECS` at topology configuration and if client has configured it then PulsarBolt will receive TickTuple. @merlimat isn't this intentional logic which will require to prevent such usecase?

Reply:
TOPOLOGY_TICK_TUPLE_FREQ_SECS is set on a per component basis and not on a topology basis

Reply:
example of usage:

https://github.com/apache/storm/blob/master/examples/storm-starter/src/jvm/org/apache/storm/starter/bolt/RollingCountBolt.java